tsdb/engine/tsm1/wal.go

```diff
@@ -709,87 +710,69 @@ func (w *WriteWALEntry) UnmarshalBinary(b []byte) error {
 		nvals := int(binary.BigEndian.Uint32(b[i : i+4]))
 		i += 4
 
-		values := make([]Value, nvals)
 		switch typ {
 		case float64EntryType:
-			for i := 0; i < nvals; i++ {
-				values[i] = FloatValue{}
-			}
-		case integerEntryType:
-			for i := 0; i < nvals; i++ {
-				values[i] = IntegerValue{}
-			}
-		case booleanEntryType:
-			for i := 0; i < nvals; i++ {
-				values[i] = BooleanValue{}
-			}
-		case stringEntryType:
-			for i := 0; i < nvals; i++ {
-				values[i] = StringValue{}
-			}
-
-		default:
-			return fmt.Errorf("unsupported value type: %#v", typ)
-		}
-
-		for j := 0; j < nvals; j++ {
-			if i+8 > len(b) {
-				return ErrWALCorrupt
-			}
-
-			un := int64(binary.BigEndian.Uint64(b[i : i+8]))
-			i += 8
-
-			switch typ {
-			case float64EntryType:
-				if i+8 > len(b) {
+			values := make([]Value, 0, nvals)
+			for j := 0; j < nvals; j++ {
+				if i+16 > len(b) {
```

Comment:
After seeing the rest of the code, it doesn't seem like a partial read is done, as the `values` slice is never stored.